src/budget.rs

```diff
@@ -105,6 +122,15 @@ impl Budget {
         if let Some(payment) = new_payment {
             mem::replace(self, Budget::Pay(payment));
         }
+        let new_partial_witness = match self {
```

Comment:
Instead, let's rename `new_payment` above to `new_budget` and change `Some(payment)` to `Some(Budget::Pay(payment))`. That way you can add `Budget::And` as just another case in that same match statement.